src/ol_openedx_git_auto_export/ol_openedx_git_auto_export/utils.py

```diff
@@ -19,3 +31,71 @@ def get_publisher_username(course_module):
 
     user = User.objects.filter(id=user_id).first()
     return user.username if user else None
+
+
+def get_or_create_git_export_repo_dir():
+    """
+    Ensure the git export directory exists and return its path.
+    """
+    git_repo_export_dir = getattr(
+        settings, "GIT_REPO_EXPORT_DIR", "/openedx/export_course_repos"
+    )
+    if not os.path.exists(git_repo_export_dir):  # noqa: PTH110
+        # for development/docker/vagrant if GIT_REPO_EXPORT_DIR folder does not exist then create it  # noqa: E501
+        log.error(
+            "GIT_REPO_EXPORT_DIR is not available in settings, please create it first"
+        )
+        os.makedirs(git_repo_export_dir, 0o755)  # noqa: PTH103
+
+    return git_repo_export_dir
+
+
+def github_repo_name_format(course_id_str):
+    """
+    Format course ID to comply with GitHub repository naming conventions using slugify.
+
+    GitHub repository names:
+    - Can only contain alphanumeric characters
+    - And hyphens (-), underscores (_), and periods (.)
+    - Cannot start or end with a hyphen
+    - Maximum length is 100 characters
+
+    Args:
+        course_id_str (str): The course ID string to format
+
+    Returns:
+        str: GitHub-compliant repository name
+    """
+    # Replace all characters with - hyphen except alphanumeric, hyphen, underscore, and period  # noqa: E501
+    repo_name = "".join(
+        char if char.isalnum() or char in "-_." else "-" for char in course_id_str
+    ).strip("-")
```

Comment:
Can be done with RE,
```suggestion
    repo_name = re.sub(r"[^A-Za-z0-9_.-]", "-", course_id_str).strip("-")
```